src/mito2/src/worker/handle_manifest.rs

```diff
@@ -0,0 +1,195 @@
+    pub(crate) fn handle_manifest_truncate_action(
+        &self,
+        region: MitoRegionRef,
+        truncate: RegionTruncate,
+        sender: OptionOutputTx,
+    ) {
+        // Marks the region as truncating.
+        // This prevents the region from being accessed by other write requests.
+        if let Err(e) = region.set_truncating() {
+            sender.send(Err(e));
+            return;
+        }
+        // Now the region is in truncating state.
+
+        let request_sender = self.sender.clone();
+        let manifest_ctx = region.manifest_ctx.clone();
+        let version_control = region.version_control.clone();
+        let memtable_builder = region.memtable_builder.clone();
+
+        // Updates manifest in background.
+        common_runtime::spawn_bg(async move {
+            // Write region truncated to manifest.
+            let action_list =
+                RegionMetaActionList::with_action(RegionMetaAction::Truncate(truncate.clone()));
+
+            let result = manifest_ctx
+                .update_manifest(RegionState::Truncating, action_list, || {
+                    // Applies the truncate action to the region.
+                    version_control.truncate(
+                        truncate.truncated_entry_id,
+                        truncate.truncated_sequence,
+                        &memtable_builder,
+                    );
+                })
+                .await;
+
+            // Sends the result back to the request sender.
+            let truncate_result = TruncateResult {
+                region_id: truncate.region_id,
+                sender,
+                result,
+                truncated_entry_id: truncate.truncated_entry_id,
+                truncated_sequence: truncate.truncated_sequence,
+            };
+            let _ = request_sender
+                .send(WorkerRequest::Background {
+                    region_id: truncate.region_id,
+                    notify: BackgroundNotify::Truncate(truncate_result),
+                })
+                .await
+                .inspect_err(|_| warn!("failed to send truncate result"));
+        });
+    }
+
+    /// Writes region change action to the manifest and then applies it to the region in background.
+    pub(crate) fn handle_manifest_region_change(
+        &self,
+        region: MitoRegionRef,
+        change: RegionChange,
+        sender: OptionOutputTx,
+    ) {
+        // Marks the region as altering.
+        if let Err(e) = region.set_altering() {
+            sender.send(Err(e));
+            return;
+        }
+
+        // Now the region is in altering state.
+        common_runtime::spawn_bg(async move {
+            let new_meta = change.metadata.clone();
+            let action_list = RegionMetaActionList::with_action(RegionMetaAction::Change(change));
+
+            let result = region
+                .manifest_ctx
+                .update_manifest(RegionState::Altering, action_list, || {
```

Comment:
Refer to `alter_region_schema()`.
https://github.com/GreptimeTeam/greptimedb/blob/3e9bda326717392945e1b7b4c41864df3ce45517/src/mito2/src/worker/handle_alter.rs#L129-L134